### PR TITLE
Use poll instead of select

### PR DESCRIPTION
**Pull request by kghost**
There is a limitation of `select` that it is unable to poll fd larger than 1023.

### CODE REVIEW

File: src/inet/InetLayer.cpp [hunk 1]
```diff
@@ -1490,34 +1482,22 @@ void InetLayer::PrepareSelect(int& nfds, fd_set *readfds, fd_set *writefds, fd_s
...
- *  @param[in]    writefds     A pointer to the set of write file descriptors.
- *
- *  @param[in]    exceptfds    A pointer to the set of file descriptors with
- *                             errors.
- *
+ *  @param[in]  pollFDs     The result of polled FDs
+ *  @param[in]  numPollFDs  The number of fds in the fd set
  */
-void InetLayer::HandleSelectResult(int selectRes, fd_set *readfds, fd_set *writefds, fd_set *exceptfds)
+void InetLayer::HandleSelectResult(const std::array<struct pollfd, WEAVE_CONFIG_MAX_POLL_FDS> & pollFDs, int numPollFDs)
```
Comment (gerickson) [inline src/inet/InetLayer.cpp:1488]: NAK. Weave coding style precludes the use of STL in the core of the library.

File: src/inet/InetLayer.h [hunk 2]
```diff
@@ -284,8 +284,8 @@ class NL_DLL_EXPORT InetLayer
 #endif // INET_CONFIG_PROVIDE_OBSOLESCENT_INTERFACES
 
 #if WEAVE_SYSTEM_CONFIG_USE_SOCKETS
-    void PrepareSelect(int& nfds, fd_set *readfds, fd_set *writefds, fd_set *exceptfds, struct timeval& sleepTime);
-    void HandleSelectResult(int selectRes, fd_set *readfds, fd_set *writefds, fd_set *exceptfds);
+    void PrepareSelect(std::array<struct pollfd, WEAVE_CONFIG_MAX_POLL_FDS> & pollFDs, int& numPollFDs, int& timeoutMS);
```
Comment (gerickson) [inline src/inet/InetLayer.h:287]: NAK. Weave coding style precludes the use of STL in the core of the library.

File: src/inet/InetLayer.h [hunk 3]
```diff
@@ -284,8 +284,8 @@ class NL_DLL_EXPORT InetLayer
...
+    void HandleSelectResult(const std::array<struct pollfd, WEAVE_CONFIG_MAX_POLL_FDS> & pollFDs, int numPollFDs);
```
Comment (gerickson) [inline src/inet/InetLayer.h:288]: NAK. Weave coding style precludes the use of STL in the core of the library.

File: src/inet/InetLayerBasis.cpp [hunk 4]
```diff
@@ -32,57 +32,54 @@ namespace Inet {
...
- *  @param[in]    writefds  A pointer to the set of writable file descriptors.
- *
- *  @param[in]    exceptfds  A pointer to the set of file descriptors with errors.
- *
+ *  @param[in]      socket      The file descriptor for which the bit is being set.
+ *  @param[in,out]  pollFDs     The fd set which is going to be polled
+ *  @param[in,out]  numPollFDs  The number of fds in the fd set
  */
-void SocketEvents::SetFDs(int socket, int& nfds, fd_set *readfds, fd_set *writefds, fd_set *exceptfds)
+void SocketEvents::SetFDs(int socket, std::array<struct pollfd, WEAVE_CONFIG_MAX_POLL_FDS> & pollFDs, int& numPollFDs)
```
Comment (gerickson) [inline src/inet/InetLayerBasis.cpp:39]: NAK. Weave coding style precludes the use of STL in the core of the library.

**Review by robszewczyk — APPROVED** [review]

**Pull request merged**